CMakeLists.txt

```diff
@@ -267,6 +267,12 @@ IF(${VERSION_LIST_SIZE} GREATER 0)
 		SET(VERSION_STRING_GIT "${MAJOR_VERSION_GIT}.${MINOR_VERSION_GIT}.${PATCH_VERSION_GIT}")
 	ENDIF()
 
+        IF("${VERSION_STRING_GIT}" MATCHES "v[0-9]+\\.[0-9]+\\.[0-9]+-[A-Za-z0-9]+")
```

Comment:
The `^v` should be optional in this string. Also, it could handle SemVer completely, who knows if someone might patch in a build/metadata suffix.